### PR TITLE
Update library_idbfs.js

### PR DESCRIPTION
**Pull request by biesie**
Chrome Browser limits IDBFS record size to 133MB for each IDBFS entry.
To store permanently and locally bigger files they need to be splitted into several parts.
This adds 
IDBFS.DB_MAXRECORDSIZE:133000000
IDBFS.DB_PARTSSUFFIX:'.idbfsp.' ( suffix added to partial finele entries)
Modifies functions 
IDBFS.loadRemoteEntry
IDBFS.storeRemoteEntry
IDBFS.removeRemoteEntry
IDBFS.getRemoteSet
So that a unique file entry bigger than 133MB, is seen as one file by the local filesystem, while its is seen as several files by the DB (with extension <DB_PARTSSUFFIX>)

### CODE REVIEW

File: src/library_idbfs.js [hunk 1]
```diff
@@ -128,8 +131,20 @@ mergeInto(LibraryManager.library, {
             if (!cursor) {
               return callback(null, { type: 'remote', db: db, entries: entries });
             }
-
-            entries[cursor.primaryKey] = { timestamp: cursor.key };
+
```
Comment (kripken) [inline src/library_idbfs.js]: please use 2 spaces for indentation, and not tabs or a different number of spaces.

File: src/library_idbfs.js [hunk 2]
```diff
@@ -128,8 +131,20 @@ mergeInto(LibraryManager.library, {
             if (!cursor) {
               return callback(null, { type: 'remote', db: db, entries: entries });
             }
-
-            entries[cursor.primaryKey] = { timestamp: cursor.key };
+			
+			var regex = new RegExp( IDBFS.DB_PARTSSUFFIX+'(\\d)(\\d)');
+			var match=cursor.primaryKey.match(regex);
```
Comment (kripken) [inline src/library_idbfs.js]: convention is spaces before and after `=`

File: src/library_idbfs.js [hunk 3]
```diff
@@ -128,8 +131,20 @@ mergeInto(LibraryManager.library, {
...
+			if (match)
```
Comment (kripken) [inline src/library_idbfs.js]: our convention is `{` on the line with the `if`

File: src/library_idbfs.js [hunk 4]
```diff
@@ -195,28 +210,190 @@ mergeInto(LibraryManager.library, {
...
+			};
+			req.onerror = function(e) {
+				callback(this.error);
+				e.preventDefault();
+			};
+		}
+		var req = store.get(path);
+		req.onsuccess = function(event) 
+		{
+			if (typeof event.target.result === 'undefined' )
```
Comment (kripken) [inline src/library_idbfs.js]: no space before `)`

File: src/library_idbfs.js [hunk 5]
```diff
@@ -195,28 +210,190 @@ mergeInto(LibraryManager.library, {
...
+			e.preventDefault();
+		};
     },
     removeRemoteEntry: function(store, path, callback) {
-      var req = store.delete(path);
-      req.onsuccess = function() { callback(null); };
-      req.onerror = function(e) {
+		function remoteDelete(store,path,index,total,callback)
+		{
+			var addstr=(index>=0?IDBFS.DB_PARTSSUFFIX+index+total:'');
```
Comment (kripken) [inline src/library_idbfs.js]: spaces around all parts of `? :` please (so, `A ? B : C`)

File: src/library_idbfs.js [hunk 6]
```diff
@@ -195,28 +210,190 @@ mergeInto(LibraryManager.library, {
...
+			  var req = store.delete(path+addstr);
```
Comment (kripken) [inline src/library_idbfs.js]: spaces before and after `+` please

File: src/library_idbfs.js [hunk 7]
```diff
@@ -195,28 +210,190 @@ mergeInto(LibraryManager.library, {
...
+				{
+					callback(null); 
+				}
+			  };
+			  req.onerror = function(e) {
+				callback(this.error);
+				e.preventDefault();
+			  };
+		}
+/*		function remoteExists(store,path,cbk)
```
Comment (kripken) [inline src/library_idbfs.js]: please remove commented-out code

**Comment by ndarilek** [comment]
Are the formatting issues the only blockers from getting this in?

Not sure if this is a resolution for #7050. @biesie, do you need help resolving these? Happy to do what I can to get this merged.

Thanks!

**Comment by kripken** [comment]
> Are the formatting issues the only blockers from getting this in?

I didn't do a full technical review here yet - it will be much easier to do so after the formatting is fixed.

File: src/library_idbfs.js (hunk 7 above)
Comment (biesie) [reply src/library_idbfs.js]: Hi, Thanks for formatting the code.
Sorry, I am new on GitHub and not familliar with the flow. Are you expecting something from me to validate?
Regards.

Comment (kripken) [reply src/library_idbfs.js]: No problem. Please fix the formatting, and make a commit with that, and then push that commit to the branch of this PR. Then the commit will show up in the PR and I can review it here.

**Pull request closed without merging**